BlueDBOnDisk/src/main/java/org/bluedb/disk/BlueCollectionOnDiskBuilder.java

```diff
@@ -0,0 +1,42 @@
+package org.bluedb.disk;
+
+import java.io.Serializable;
+
+import org.bluedb.api.BlueCollectionBuilder;
+import org.bluedb.api.exceptions.BlueDbException;
+import org.bluedb.api.keys.BlueKey;
+import org.bluedb.disk.collection.BlueCollectionOnDisk;
+
+public class BlueCollectionOnDiskBuilder<T extends Serializable> implements BlueCollectionBuilder<T>{
+
+	private final BlueDbOnDisk db;
+	private final Class<T> valueType;
+	private final Class<? extends BlueKey> requestedKeyType;
+	String name;

```

Comment:
Name is a required parameter so I'm wondering if we should set it in the constructor instead of having the `withName` method. As it stands now, if they don't call `withName` is it going to just throw an exception? I'd rather make it more obvious that you have to give it a name.

Reply:
The main reason I want a builder is to take out optional things (like the varargs classes or the future segment size options).  I suppose we could require a name when you instantiate the builder to prevent run time exceptions.

Do you want collectionBuilder(...) to have all required parameters and optional parameters added in other builder methods?

Reply:
That's what I was thinking. I don't think we'll really be adding required things from now on, just optional configuration and stuff. I think I'd rather have the extra parameter than for the user to have to remember to set the name.